src/control/cmd/dmg/pretty/pool.go

```diff
@@ -85,3 +86,166 @@ func PrintPoolCreateResponse(pcr *control.PoolCreateResp, out io.Writer, opts ..
+
+	// display imbalance of the most imbalanced tier
+	imbalancedTier := pool.Usage[0]
+	for ti := 0; ti < len(pool.Usage); ti++ {
+		if pool.Usage[ti].Imbalance > imbalancedTier.Imbalance {
+			imbalancedTier = pool.Usage[ti]
+		}
+	}
+	imbalanceStr := fmt.Sprintf("%d%% (%s)", imbalancedTier.Imbalance, imbalancedTier.TierName)
+
+	row := txtfmt.TableRow{
+		"Pool":      name,
+		"Size":      sizeStr,
+		"Used":      usedStr,
+		"Imbalance": imbalanceStr,
+		"Disabled":  fmt.Sprintf("%d/%d", pool.TargetsDisabled, pool.TargetsTotal),
+	}
+
+	return row
+}
+
+func printListPoolsResp(out io.Writer, resp *control.ListPoolsResp) error {
+	if len(resp.Pools) == 0 {
+		fmt.Fprintln(out, "no pools in system")
+		return nil
+	}
+
+	formatter := txtfmt.NewTableFormatter("Pool", "Size", "Used", "Imbalance", "Disabled")
+
+	var table []txtfmt.TableRow
+	for _, pool := range resp.Pools {
+		table = append(table, poolListCreateRow(pool))
+	}
+
+	fmt.Fprintln(out, formatter.Format(table))
+
+	return nil
+}
+
+func addVerboseTierUsage(row txtfmt.TableRow, usage *control.PoolTierUsage) txtfmt.TableRow {
+	row[usage.TierName+" Size"] = humanize.Bytes(usage.Size)
+	row[usage.TierName+" Used"] = humanize.Bytes(usage.Size - usage.Free)
+	row[usage.TierName+" Imbalance"] = fmt.Sprintf("%d%%", usage.Imbalance)
+
+	return row
+}
+
+func poolListCreateRowVerbose(pool *control.Pool) txtfmt.TableRow {
+	label := pool.Label
+	if label == "" {
+		label = "-"
+	}
+
+	svcReps := "N/A"
+	if len(pool.ServiceReplicas) != 0 {
+		rl := system.RanksToUint32(pool.ServiceReplicas)
+		svcReps = formatRanks(rl)
+	}
+
+	row := txtfmt.TableRow{
+		"Label":    label,
+		"UUID":     pool.UUID,
+		"SvcReps":  svcReps,
+		"Disabled": fmt.Sprintf("%d/%d", pool.TargetsDisabled, pool.TargetsTotal),
+	}
+
+	for _, tu := range pool.Usage {
+		row = addVerboseTierUsage(row, tu)
+	}
+
+	return row
+}
+
+func printListPoolsRespVerbose(out io.Writer, resp *control.ListPoolsResp) error {
+	if len(resp.Pools) == 0 {
+		fmt.Fprintln(out, "no pools in system")
+		return nil
+	}
+
+	titles := []string{"Label", "UUID", "SvcReps"}
+	for _, t := range resp.Pools[0].Usage {
+		titles = append(titles,
+			t.TierName+" Size",
+			t.TierName+" Used",
+			t.TierName+" Imbalance")
+	}
+	titles = append(titles, "Disabled")
+	formatter := txtfmt.NewTableFormatter(titles...)
+
+	var table []txtfmt.TableRow
+	for _, pool := range resp.Pools {
+		table = append(table, poolListCreateRowVerbose(pool))
+	}
+
+	fmt.Fprintln(out, formatter.Format(table))
+
+	return nil
+}
+
+func validateListPoolsResp(r *control.ListPoolsResp) error {
+	var numTiers int
+
+	for i, p := range r.Pools {
+		if p.UUID == "" {
+			return errors.Errorf("pool with index %d has no uuid", i)
+		}
+		if len(p.Usage) == 0 {
+			return errors.Errorf("pool %s has no usage info", p.UUID)
+		}
+		if numTiers != 0 && len(p.Usage) != numTiers {
+			return errors.Errorf("pool %s has %d storage tiers, want %d",
+				p.UUID, len(p.Usage), numTiers)
+		}
```

Comment:
Not a problem if so, but I'm curious if this is necessary to check. If it didn't have all the tiers, would it be OK to leave the columns empty instead?

Reply:
Without this check you would still have to iterate through all pools in the response to find the one with the highest number of storage tiers in order to set the list of titles before creating any row data. Then you could do as you suggest. As the actual number of storage tiers should be fixed at format time this check seems okay.